inception-scheduling/src/main/java/de/tudarmstadt/ukp/inception/scheduling/SchedulingService.java

```diff
@@ -85,13 +85,15 @@ private void afterExecute(Runnable aRunnable, Throwable aThrowable)
         return result;
     }
 
-    public void enqueue(Task aTask)
+    public synchronized void enqueue(Task aTask, String aOrigin)
```

Comment:
`aOrigin` -> `aReason`?

Reply:
I do not like reason too much, as it implies causality

Reply:
I think causality is given. We wouldn't start a task if there wasn't a reason for it?

Reply:
I understand that your wording might stem from Exceptions

Reply:
If you don't like reason, how about `aTrigger`? ;)

Reply:
Done